### PR TITLE
added .gitattributes file

### LINKED ISSUE #129: Do we need .gitattributes file?
Something like this:

```
.editorconfig    export-ignore
.gitattributes   export-ignore
.gitignore       export-ignore
CONTRIBUTING.md  export-ignore
```

Please use 👍  or 👎  to vote!

### PR DESCRIPTION
**Pull request by OskarStark**
Closes #129

### CODE REVIEW

File: project/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.editorconfig    export-ignore
```
Comment (Soullivaneuh) [inline project/.gitattributes]: Why not `.*`?

**Comment by Soullivaneuh** [comment]
Could we add the documentation part?

File: project/.gitattributes [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+.gitattributes   export-ignore
+.gitignore       export-ignore
+CONTRIBUTING.md  export-ignore
+
+/Resources/doc   export-ignore
```
Comment (Soullivaneuh) [inline project/.gitattributes]: This is not always this path. Pleas use `{{ docs_path }}` instead.

Example: https://github.com/sonata-project/dev-kit/blob/ae18f3ea3755f543d83b4f713b752dc66398216d/project/Makefile#L17

**Comment by greg0ire** [comment]
What about tests?

**Comment by Soullivaneuh** [comment]
> What about tests?

I would like #116 consensus before work on this.

File: project/.gitattributes (hunk 1 above)
Comment (OskarStark) [inline project/.gitattributes]: we can...

File: project/.gitattributes (hunk 2 above)
Comment (OskarStark) [inline project/.gitattributes]: done

File: project/.gitattributes (hunk 1 above)
Comment (OskarStark) [inline project/.gitattributes]: done

**Comment by OskarStark** [comment]
i got this error:

```
oskar.stark:/Volumes/development/workspaces/dev-kit (gitattributes-file)$ git rebase -i HEAD~2
}} is not a valid attribute name: project/.gitattributes:4
}} is not a valid attribute name: project/.gitattributes:4
}} is not a valid attribute name: project/.gitattributes:4
[losgelöster HEAD 4402d52] added .gitattributes file
 Date: Wed Jun 1 17:04:46 2016 +0200
 1 file changed, 4 insertions(+)
 create mode 100644 project/.gitattributes
Successfully rebased and updated refs/heads/gitattributes-file.
oskar.stark:/Volumes/development/workspaces/dev-kit (gitattributes-file)$
````

**Comment by greg0ire** [comment]
Maybe name it `.gitattributes.twig` ?

**Comment by OskarStark** [comment]
> Maybe name it .gitattributes.twig ? It will avoid git recognizing it for dev-kit

sound good, what do you think @Soullivaneuh ?

**Comment by Soullivaneuh** [comment]
@OskarStark Yes you can. In this case, the Twig variable name would be `docs_path`.

**Comment by OskarStark** [comment]
so `docs_path` instead of `{{ docs_path }}` ?

**Comment by Soullivaneuh** [comment]
@OskarStark Well, it's a Twig file! So AFAIK you still need to use `{{ }}`! :trollface:

**Comment by OskarStark** [comment]
> @OskarStark Well, it's a Twig file! So AFAIK you still need to use {{ }}! :trollface:

Exactly thats why i was asking! so now we should be fine! 👍

File: project/.gitattributes.twig [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.*               export-ignore
+CONTRIBUTING.md  export-ignore
```
Comment (Soullivaneuh) [inline project/.gitattributes.twig]: `*.md`? Except if somebody want to keep a specific `.md` file...

Comment (OskarStark) [inline project/.gitattributes.twig]: > *.md? Except if somebody want to keep a specific .md file...

i like the idea 👍

Comment (OskarStark) [inline project/.gitattributes.twig]: done

**Comment by OskarStark** [comment]
i removed the indention, because we don't know the length of the docs path....

**Comment by Soullivaneuh** [comment]
> i removed the indention, because we don't know the length of the docs path....

Hmm... or we can just specify the two cases (`Resources/doc` and `docs`).

I don't think git will cry if a ignored path does not exist, will do?

**Comment by OskarStark** [comment]
no it won't but the indention is anyway bad if we add something like `foo/bar/baz/loooonnnnggg` tomorrow, the the whole file will be modified in the diff, it is the exact same problem with the double arrow align

**Comment by greg0ire** [comment]
What's wrong with the Twig variable?

**Comment by Soullivaneuh** [comment]
> no it won't but the indention is anyway bad

Maybe, but I never seen a `.gitattributes` file without indentation. I don't know if a convention exists about that file...

**Comment by OskarStark** [comment]
> Maybe, but I never seen a .gitattributes file without indentation. I don't know if a convention exists about that file...

https://git-scm.com/book/uz/v2/Customizing-Git-Git-Attributes#Merge-Strategies

**Comment by greg0ire** [comment]
`man gitattributes`

**Comment by OskarStark** [comment]
> What's wrong with the Twig variable?

Nothing i thought i should change `{{ docs_path }}` to `docs_path` without the curly braces... that confused me a few seconds 😄

**Comment by greg0ire** [comment]
>     What's wrong with the Twig variable?

No I meant why not use it, but I get it know, it was all about alignment

**Comment by OskarStark** [comment]
for example when we use this:

```
foo                      export-ignore
bar                      export-ignore
{{ docs_path}}     export-ignore
```

this would be the result:
```
foo                      export-ignore
bar                      export-ignore
/Resources/docs    export-ignore
```

you know what i mean?

**Comment by greg0ire** [comment]
> you know what i mean?

Yes. Headaches ahead if we do that, as you explained.

**Comment by Soullivaneuh** [comment]
@OskarStark Looks good. Can you try it on a one of your forked sonata project?

After that, I would like @rande opinion for this before merge because I got a refusal from him last time and he broke my heart. :broken_heart: :stuck_out_tongue:

**Comment by greg0ire** [comment]
please

**Comment by rande** [comment]
@Soullivaneuh You should also ask the community ;)

**Comment by Soullivaneuh** [comment]
> @Soullivaneuh You should also ask the community ;)

Indeed, we might start a pool. I'll look for this week-end or next week. :+1:

**Comment by Soullivaneuh** [comment]
> You should also ask the community ;)

After all, @OskarStark already start a vote.

@rande @sonata-project/contributors: https://twitter.com/soullivaneuh/status/738499577775566848

**Comment by greg0ire** [comment]
Well the poll is overwhelmingly for this change, so RTM

**Comment by Soullivaneuh** [comment]
@rande I let you merge this one.

**Comment by Soullivaneuh** [comment]
@rande Ok to merge?

I would like to merge it as the same sime as the pending others.

I just need your approbation.

**Comment by Soullivaneuh** [comment]
Found this: https://github.com/doctrine/collections/pull/79

Seems to be adopted by Doctrine too.

**Comment by greg0ire** [comment]
@Soullivaneuh : It has been two weeks, I don't think @rande will hold a grudge against us if we merge this.

**Comment by Soullivaneuh** [comment]
> @Soullivaneuh : It has been two weeks, I don't think @rande will hold a grudge against us if we merge this.

I'm too scary. :trollface:

**Comment by rande** [comment]
I am against it if you exclude the Docs folder. Documentation is code too ...

**Comment by Soullivaneuh** [comment]
> I am against it if you exclude the Docs folder. Documentation is code too ...

RST format is not the best way to read, online documentation is...

@OskarStark Can you remove doc from `.gitattributes` for now? We can merge the rest and open a new issue to vote about that.

**Comment by OskarStark** [comment]
> @OskarStark Can you remove doc from .gitattributes for now? We can merge the rest and open a new issue to vote about that.

done @Soullivaneuh

**Comment by Soullivaneuh** [comment]
Thanks @OskarStark, this will be merged soon.

If you want to, you could open an issue to discuss about documentation ignore with vote.

**Comment by OskarStark** [comment]
i agree with @rande that documentation could be helpful in some rare cases, so for me this is fine as a first step!

**Comment by core23** [comment]
Maybe we should remove `phpunit.xml.dist` and `Makefile` too

**Comment by OskarStark** [comment]
sounds reasonable...

@greg0ire @Soullivaneuh what do you think?

File: project/.gitattributes.twig [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.* export-ignore
+*.md export-ignore
```
Comment (core23) [inline project/.gitattributes.twig]: When removing all `Upgrade*.md` and `Readme.md` files, we should also remove the docs.

**Comment by greg0ire** [comment]
I agree!

**Comment by Soullivaneuh** [comment]
> Maybe we should remove `phpunit.xml.dist` and `Makefile` too

Agree.

> When removing all `Upgrade*.md` and `Readme.md` files, we should also remove the docs.

Quite agree but let's discuss about this on another PR/issue! :+1:

**Comment by OskarStark** [comment]
but, if we remove the `phpunit.xml.dist` shouldn't we remove the tests, too ❓

**Comment by Soullivaneuh** [comment]
> but, if we remove the `phpunit.xml.dist` shouldn't we remove the tests, too ❓ 

We don't care on this case, it's applies only for tarball, not clones.

**Comment by core23** [comment]
> but, if we remove the `phpunit.xml.dist` shouldn't we remove the tests, too ❓ 

We don't need the phpunit config outside the bundle, but some bundles depends on the `Tests/*.php` classes. Refs https://github.com/sonata-project/dev-kit/issues/179

**Comment by Soullivaneuh** [comment]
> We don't need the phpunit config outside the bundle, but some bundles depends on the `Tests/*.php` classes. Refs https://github.com/sonata-project/dev-kit/issues/179

Yeah, but this should be removed, shouldn't be?

File: project/.gitattributes.twig (hunk 4 above)
Comment (core23) [inline project/.gitattributes.twig]: >> When removing all `Upgrade*.md` and `Readme.md` files, we should also remove the docs.

> Quite agree but let's discuss about this on another PR/issue! :+1: 

Then we should remove this entry.

**Comment by core23** [comment]
> Yeah, but this should be removed, shouldn't be?

Yes, but we can't remove them in all minor versions.

File: project/.gitattributes.twig (hunk 4 above)
Comment (Soullivaneuh) [inline project/.gitattributes.twig]: @rande Talk about keeping the docs folder, not the md files. Plus, the `README.md` file is just a generic one with a link pointing to the online documentation.

So let's keep as is for now and launch a vote for docs folders.

**Comment by Soullivaneuh** [comment]
> Yes, but we can't remove them in all minor versions.

Already discussed on the concerned issue: This have to be fixed internally on **minor** versions. We don't care about external usage because it should not be used at all.

**Comment by core23** [comment]
We can now remove the `Tests` folder, it is excluded from the autoloader in all composer projects.

**Comment by core23** [comment]
Ping @OskarStark

**Comment by OskarStark** [comment]
done @core23

**Comment by OskarStark** [comment]
I did everything, can you please merge this @Soullivaneuh ?

**Comment by OskarStark** [comment]
Can we please merge this @Soullivaneuh ?

**Comment by OskarStark** [comment]
ping @Soullivaneuh

**Comment by core23** [comment]
Can we merge this @Soullivaneuh ?

**Comment by core23** [comment]
Ping @Soullivaneuh

**Pull request merged**